File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        
        count_w = blocks[0:k].count('W')
        print(count_w)
        l = 0
        mini =count_w
        for r in range(k,len(blocks)):
            l+=1
            if blocks[l-1] == 'W':
                count_w-=1
            if blocks[r] == 'W':
                count_w+=1
            mini = min(count_w, mini)
        return mini

File: 2463-minimum-recolors-to-get-k-consecutive-black-blocks/minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        
        int white = 0;
        int minWhite = Integer.MAX_VALUE;
        for(int i=0;i<k;i++){
            if(blocks.charAt(i) == 'W'){
                white++;
            }
        }
        minWhite = Math.min(white, minWhite);

        for(int j=k;j<blocks.length();j++){
            if(blocks.charAt(j-k) == 'W'){
                white--;
            }
            if(blocks.charAt(j) == 'W'){
                white++;

            }
        minWhite = Math.min(white, minWhite);
        }
        return minWhite;



    }
}